src/sqlfluff/dialects/dialect_bigquery.py

```diff
@@ -417,8 +417,33 @@ class LiteralCoercionSegment(BaseSegment):
     )
 
 
+@bigquery_dialect.segment(replace=True)
+class ObjectReferenceSegment(ansi_dialect.get_segment("ObjectReferenceSegment")):  # type: ignore
+    """A reference to an object."""
+
+    pass
+
+
+@bigquery_dialect.segment(replace=True)
+class ColumnReferenceSegment(ObjectReferenceSegment):
+    """A reference to column, field or alias."""
+
+    type = "column_reference"
+
+    def extract_possible_references(self, level: int):
+        """Extract possible references of a given level."""
+        refs = list(self.iter_raw_references())
+        if level == 3 and len(refs) >= 3:
```

Comment:
I think I need a little help (or comment) to understand which level corresponds to which reference, column / field / alias?

Reply:
If each level has a meaning, could it be turned into a named enumerator?

Reply:
I had the same thought -- I'll look into it for this PR. Until I do that, this [section of`dialect_ansi.py`](https://github.com/sqlfluff/sqlfluff/blob/master/src/sqlfluff/dialects/dialect_ansi.py#L612L614) documents the meaning of the values.